### PR TITLE
Fix `NOTIFY_USER` with another event issue

### PR DESCRIPTION
**Pull request by Air-t**
* This fixes the issue when too `NOTIFY_USER` event is added
* with any other event to the same webhook. In this case
* `NOTIFY_USER` will not be triggered. Fix prevents to save
* multiple events along with `NOTIFY_USER` event.

Resolves https://github.com/saleor/saleor/issues/13577

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are either absent or guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migrations are either absent or optimized for zero downtime
* [ ] The changes are covered by test cases

### CODE REVIEW

File: saleor/graphql/webhook/mixins.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+class NotifyUserEventValidationMixin:
+    @classmethod
+    def validate_events(cls, events):
+        # NOTIFY_USER needs to be the only one event registered per webhook.
+        # This solves issue temporarily. NOTIFY_USER will be deprecated in the future.
+        if WebhookEventAsyncType.NOTIFY_USER in events and len(events) > 1:
+            raise ValidationError(
+                {
+                    "async_events": ValidationError(
+                        "Notify User event has to be the only one event per webhook.",
```
Comment (maarcingebala) [inline saleor/graphql/webhook/mixins.py:16]: I would rephrase a bit to something like this:
"The `NOTIFY_USER` webhook cannot be combined with other events."

File: saleor/graphql/webhook/mixins.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+                        code=WebhookErrorCode.INVALID.value,
```
Comment (maarcingebala) [inline saleor/graphql/webhook/mixins.py:17]: I also think this is a very specific error, and providing a dedicated error code might be better. Then we could open a ticket in the dashboard to handle this specific error and provide a correct error message. Otherwise, the INVALID code may result in a too-generic error message, not clear to the user.

File: saleor/graphql/webhook/tests/mutations/test_webhook_create.py [hunk 3]
```diff
@@ -322,3 +322,33 @@ def test_webhook_create_invalid_custom_headers(app_api_client):
...
+        query,
+        variables=variables,
+    )
+    content = get_graphql_content(response)
+    # then
+    data = content["data"]["webhookCreate"]
+    assert not data["webhook"]
+    error = data["errors"][0]
+    assert error["field"] == "asyncEvents"
+    assert error["message"] == (
```
Comment (maarcingebala) [inline saleor/graphql/webhook/tests/mutations/test_webhook_create.py:352]: If we had a specific error code for this error, we could rely on checking the code and not the message in the test (which is a bit better as the code never or rarely changes).

File: saleor/graphql/webhook/mixins.py (hunk 2 above)
Comment (Air-t) [reply saleor/graphql/webhook/mixins.py:17]: do we have any convention for bug fix error codes? Or just something like `COMBINE_FORBBIDEN` is enough?

Comment (maarcingebala) [reply saleor/graphql/webhook/mixins.py]: You can see examples in `error_codes.py` files, some codes are simple and some are very specific. Here I would go with something descriptive like: `INVALID_NOTIFY_WITH_SUBSCRIPTION` or `CANNOT_COMBINE_NOTIFY_AND_SUBSCRIPTION`.

**Review by 8r2y5 — APPROVED** [review]

**Review by szczecha — APPROVED** [review]
